genome_designer/genome_finish/assembly.py

```diff
@@ -1,169 +1,379 @@
-                SeqIO.write([seq_record], fh, 'fasta')
-
-            add_dataset_to_entity(
-                    contig,
-                    'contig_fasta',
-                    Dataset.TYPE.REFERENCE_GENOME_FASTA,
-                    filesystem_location=dataset_path)
+        pickle.dump(assembly_metadata, fh)
+
+    cmd = ' '.join([
+            VELVETH_BINARY,
+            data_dir,
+            str(velvet_opts['velveth']['hash_length'])] +
+            ['-' + key + ' ' + str(velvet_opts['velveth'][key])
+             for key in velvet_opts['velveth']
+             if key not in ['hash_length']] +
+            ['-bam', '-shortPaired', sv_indicants_bam])
+    print 'velveth cmd:', cmd
+    subprocess.check_call(cmd, shell=True, executable=settings.BASH_PATH)
+
+    ins_length = velvet_opts['velvetg']['ins_length']
+    # exp_cov = velvet_opts['velvetg']['exp_cov']
+    ins_length_sd = velvet_opts['velvetg']['ins_length_sd']
+    cov_cutoff = velvet_opts['velvetg']['cov_cutoff']
+    min_contig_lgth = velvet_opts['velvetg']['min_contig_lgth']
+
+    arg_list = [
+            VELVETG_BINARY,
+            data_dir,
+            '-ins_length', str(ins_length),
+            # '-exp_cov', str(exp_cov),
+            '-ins_length_sd', str(ins_length_sd),
+            '-cov_cutoff', str(cov_cutoff),
+            # '-max_coverage', str(3 * exp_cov),
+            '-min_contig_lgth', str(min_contig_lgth),
+            '-read_trkg', 'yes']
+
+    cmd = ' '.join(arg_list)
+    print 'velvetg cmd:', cmd
+    subprocess.check_call(cmd, shell=True, executable=settings.BASH_PATH)
+
+    # Collect resulting contigs fasta
+    contigs_fasta = os.path.join(data_dir, 'contigs.fa')
+    contig_files.append(contigs_fasta)
+
+    for seq_record in SeqIO.parse(contigs_fasta, 'fasta'):
+
+        contig_node_number = int(
+                contig_number_pattern.findall(seq_record.description)[0])
+        contig_label = 'Contig_' + str(contig_node_number)
+
+        # Create an insertion model for the contig
+        contig = Contig.objects.create(
+                label=contig_label,
+                parent_reference_genome=reference_genome,
+                experiment_sample_to_alignment=(
+                        sample_alignment))
+        contig_list.append(contig)
+
+        contig.metadata['coverage'] = float(
+                seq_record.description.rsplit('_', 1)[1])
+        contig.metadata['timestamp'] = timestamp
+        contig.metadata['node_number'] = contig_node_number
+        contig.metadata['assembly_dir'] = data_dir
+
+        contig.ensure_model_data_dir_exists()
+        dataset_path = os.path.join(contig.get_model_data_dir(),
+                'fasta.fa')
+
+        with open(dataset_path, 'w') as fh:
+            SeqIO.write([seq_record], fh, 'fasta')
 
+        add_dataset_to_entity(
+                contig,
+                'contig_fasta',
+                Dataset.TYPE.REFERENCE_GENOME_FASTA,
+                filesystem_location=dataset_path)
```

Comment:
I think you might need to use clean_filesystem_location(). Search around to see how we use this. Unfortunately maybe inconsistent.

The point is that filesystem_location should store a path relative to `MEDIA_ROOT`.